### PR TITLE
Import 90% of machine code in crc

### PR DESCRIPTION
**Pull request by guillaumerose**
Only drivers (rpc, driver interface, hyperkit and libvirt structs) and state are left in code-ready/machine.

It will be easier to refactor machine code: better tests in the future, etc. All the persistence is now done in crc side. code-ready/machine is stateless and also all drivers.

will enable https://github.com/code-ready/machine/pull/54

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -99,3 +99,5 @@ replace (
 	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.19.0
 	k8s.io/sample-controller => k8s.io/sample-controller v0.19.0
 )
+
+replace github.com/code-ready/machine => github.com/guillaumerose/machine v0.0.0-20201224094349-747e5fa2ff83
```
Comment (gbraad) [inline go.mod:103]: what still remains there?

File: pkg/drivers/fakedriver/fakedriver.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
+package fakedriver
```
Comment (gbraad) [inline pkg/drivers/fakedriver/fakedriver.go:1]: Used by `pkg/libmachine/host/host_test.go`

File: pkg/libmachine/host/migrate_test.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+				},
+			},
+			expectedMigrationError: nil,
+		},
+		{
+			description: "Config version 4 (from the FUTURE) on disk",
+			rawData: []byte(`{
+    "ConfigVersion": 4,
+    "Driver": {"MachineName": "default"},
+    "DriverName": "virtualbox",
```
Comment (gbraad) [inline pkg/libmachine/host/migrate_test.go:64]: we never had such a thing ... so let's not suggest we do ;-)

File: pkg/libmachine/host/migrate_test.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    "HostOptions": {
+        "Driver": "",
+        "Memory": 0,
+        "Disk": 0,
+        "AuthOptions": {
+            "StorePath": "/Users/nathanleclaire/.code-ready/machine/machines/default"
```
Comment (gbraad) [inline pkg/libmachine/host/migrate_test.go:70]: `.code-ready` ?

File: pkg/drivers/fakedriver/fakedriver.go (hunk 2 above)
Comment (guillaumerose) [reply pkg/drivers/fakedriver/fakedriver.go:1]: Removed. It was not bringing much info.

File: go.mod (hunk 1 above)
Comment (guillaumerose) [reply go.mod:103]: not much https://github.com/code-ready/machine/pull/54

the RPC, driver interface and some common structs. Few lines.

File: pkg/libmachine/host/migrate_test.go (hunk 3 above)
Comment (guillaumerose) [reply pkg/libmachine/host/migrate_test.go]: I rewrote the test, it's now gone.

File: pkg/libmachine/host/migrate_test.go (hunk 4 above)
Comment (guillaumerose) [reply pkg/libmachine/host/migrate_test.go]: test removed

**Comment by guillaumerose** [comment]
Ready for an other round of review :)

**Review by cfergeau — COMMENTED** [review]
I think I'd keep all of `pkg/drivers` in `machine`, but no strong preferenc either way.

**Comment by guillaumerose** [comment]
I also hesitated for hyperv driver but at the end, it is much easier to have machine reduced to the strict minimum. The major benefits is the ability to share code between preflights and the driver (how we invoke powershell, etc).

**Comment by guillaumerose** [comment]
rebased!

**Review by gbraad — APPROVED** [review]
Addreessed the issues I had, and it uses crc/logging instead of libmachine/log

File: pkg/crc/machine/machine.go [hunk 5]
```diff
@@ -9,9 +9,9 @@ import (
 	"github.com/code-ready/crc/pkg/crc/logging"
 	"github.com/code-ready/crc/pkg/crc/machine/bundle"
 	"github.com/code-ready/crc/pkg/crc/network"
-	"github.com/code-ready/machine/libmachine"
+	"github.com/code-ready/crc/pkg/libmachine"
+	"github.com/code-ready/crc/pkg/libmachine/host"
 	"github.com/code-ready/machine/libmachine/drivers"
-	"github.com/code-ready/machine/libmachine/host"
 	"github.com/code-ready/machine/libmachine/log"
```
Comment (gbraad) [inline pkg/crc/machine/machine.go:15]: What would prevent this to become `log "github.com/code-ready/crc/pkg/crc/logging" ?

Comment (guillaumerose) [reply pkg/crc/machine/machine.go:15]: Not much. I have a PR almost ready after this one to remove the machine logging and use traditional logrus.

Comment (gbraad) [reply pkg/crc/machine/machine.go:15]: 👍

Comment (gbraad) [reply pkg/crc/machine/machine.go:15]: Do you want to address that, or push another PR to finalize? I'd rather not keep PRs hanging around, as this causes unnecessary difficult rebases, etc.

Comment (guillaumerose) [reply pkg/crc/machine/machine.go:15]: I have to change machine before propagating it here in crc :/

**Pull request merged**